pkg/granted/credentialsprocess.go

```diff
@@ -0,0 +1,103 @@
+package granted
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/smithy-go"
+	"github.com/common-fate/granted/pkg/cfaws"
+	"github.com/fatih/color"
+	"github.com/urfave/cli/v2"
+)
+
+/*
+* AWS Creds consumed by credential_process must adhere to this schema
+* https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html
+ */
+type AWSCredsStdOut struct {
+	Version         int    `json:"Version"`
+	AccessKeyID     string `json:"AccessKeyId"`
+	SecretAccessKey string `json:"SecretAccessKey"`
+	SessionToken    string `json:"SessionToken"`
+	Expiration      string `json:"Expiration"`
+}
+
+var CredentialsProcess = cli.Command{
+	Name:   "credentialsprocess",
+	Usage:  "",
+	Hidden: true,
+	Flags:  []cli.Flag{&cli.StringFlag{Name: "profile", Required: true}, &cli.StringFlag{Name: "url"}},
+	Action: func(c *cli.Context) error {
+
+		url := c.String("url")
+		profileName := c.String("profile")
+
+		if url == "" {
```

Comment:
We should handle this better, as this assumes everyone who is using the `granted credentialsprocess` command is also using Granted Approvals. If the URL isn't provided, we should simply print an error message without a request URL.

Reply:
Good reccomendation @chrnorm. URL handling [amended here](https://github.com/common-fate/granted/commit/1cceea287a2f6aad32a820911d0b1bff8ca0ccaf), it now will only throw an error at the URL generation step (if still found to be empty string ""). This allows it to dual-function without an approvals deployment